Fix App.framework path in Podfile

jmagman opened the pull request:
## Description

There was a bug with the path logic in creating the Debug fake `static const int Moo = 88;` App.framework.  Fix the path.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/63176

## Tests

Add a check to make sure the App.framework is really being embedded in the host app.
Added a `flutter clean` and `flutter pub get` to the integration test.  Before, it was using the App.framework from a previous step building it from the module, which doesn't prove it can be generated from the host app without running `flutter build` in the module first.

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

jmagman committed:
Fix App.framework path in Podfile

christopherfujino reviewed (approved):
I'll LGTM anything with less ruby

jmagman commented:
I don't think this test is complete.

xster commented:
LGTM. Thanks for adding tests

jmagman commented:
I was wrong, we still need the fake App.framework.  I forgot about the `-framework App` in the `OTHER_LDFLAGS` Pods xcconfig.
I updated this to just fix the path.

Also, I added a `flutter clean` and `flutter pub get` to the integration test.  Before, it was using the App.framework from a previous step building it from the module, which doesn't prove it can be generated from the host app without running `flutter build` in the module first.

szotp commented:
LGTM and works on my machine
